### ISSUE TITLE
Shadow Warrior 2 (324800)

### ISSUE THREAD

**Issue by springwurm** [ValveSoftware/Proton#4031]
# Compatibility Report
- Name of the game with compatibility issues: Shadow Warrior 2
- Steam AppID of the game: 324800

## System Information
- GPU: AMD Radeon RX 580
- Driver/LLVM version: Mesa 20.1.2
- Distro version: Arch Linux
- Link to full system information report as [Gist](https://gist.github.com/springwurm/10b9ae03c39ad871ce714f01ea4bc384):
- Proton version: 5.0-9

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

## Symptoms
- Videos won't play

## Reproduction
- Click Play > Main Menu > Extras > Videos > Select a Video.

## Logs
- [steam-324800.log](https://github.com/ValveSoftware/Proton/files/4842082/steam-324800.log)

**Comment by kisak-valve** [comment]
Hello @springwurm, looks like this game would benefit from support for Media Foundation maturing.

**Comment by btegs** [comment]
@springwurm - Have you tried  Proton-GE from https://github.com/GloriousEggroll/proton-ge-custom/releases? Sometimes the GE version has better playback for Media Foundation stuff due to it not being blacklisted like mainline Proton.

**Comment by springwurm** [comment]
> @springwurm - Have you tried Proton-GE from https://github.com/GloriousEggroll/proton-ge-custom/releases? Sometimes the GE version has better playback for Media Foundation stuff due to it not being blacklisted like mainline Proton.

yes I tried the Proton-5.11-GE-1-MF release, but there was no difference. `<Workaround removed by moderator>`

Proton-5.11-GE-1-MF Logs: [steam-324800_Proton-5.11-GE-1-MF.log](https://github.com/ValveSoftware/Proton/files/4842955/steam-324800_Proton-5.11-GE-1-MF.log)

**Comment by kisak-valve** [comment]
Hello @springwurm, the workaround you mentioned is legally problematic and has been removed.

**Comment by springwurm** [comment]
to be clear, i used the workaround only for testing. After that, i removed the Wine/Proton prefix.

The game works so far without issues.

**Comment by btegs** [comment]
> Hello @springwurm, the workaround you mentioned is legally problematic and has been removed.

What did he do wrong? Was it the manual usage of mf-install or just using GloriousEggroll's version?

**Comment by kisak-valve** [comment]
Hello @btegs, please see https://github.com/ValveSoftware/Proton/issues/2594#issuecomment-554788751 and the comment immediately before it.

**Comment by btegs** [comment]
> Hello @btegs, please see [#2594 (comment)](https://github.com/ValveSoftware/Proton/issues/2594#issuecomment-554788751) and the comment immediately before it.

Considering that this is supposed to be a FLOSS project on an open platform, I totally understand not including proprietary executables. It would be nice if there was some ffmpeg and/or libav implementation to do the same.

**Comment by tgurr** [comment]
Keeps crashing for since the latest Proton 6.3-1 release, while it works beautiful with Proton 5.13-6.

Proton 6.3-1 Log: [steam-324800_Proton-6.3-1.log](https://github.com/ValveSoftware/Proton/files/6246305/steam-324800_Proton-6.3-1.log)

Shadow Warrior 2 is always the first game I try when there are new Proton releases as it starts up blazingly fast and has no interrupting videos/intros to go through first.

Note: I'm seeing the same issue with Proton-6.1-GE-2 (works) vs. Proton-6.5-GE-1 (crashing).

**Comment by kisak-valve** [comment]
Hello @tgurr, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by tgurr** [comment]
> Hello @tgurr, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

@kisak-valve sure, sorry, should've thought of it in the first place, here it is: https://gist.github.com/tgurr/b45138598ffb973686007b426733dca6

**Comment by alasky17** [comment]
@tgurr Could you be more specific about where/when it crashes?  I'm able to load into an existing save point on 6.3-1 and play for ~1 min without seeing a crash.

**Comment by tgurr** [comment]
@alasky17 interesting, for me it crashes after I hit loading a savegame, right during the loading screen, so with 6.3-1 I was never able to actually get ingame.

**Comment by kodatarule** [comment]
[steam-324800.log](https://github.com/ValveSoftware/Proton/files/6251309/steam-324800.log)
The game crashes upon loading a level. Both with experimental and proton 6.3-1

**Comment by ipr** [comment]
Crash immediately in main menu.
[steam-324800.log](https://github.com/ValveSoftware/Proton/files/6835205/steam-324800.log)

**Comment by Sanaki** [comment]
Been playing 40 hours on 6.3-6 so far with no crashes after adding the `-novideos` flag to launch options.

That said, I came here hoping someone mentioned the CPU spikes causing intermittent slowdown. From what I can tell the game's simply notorious for heavy CPU usage even on Windows, but looking at the same scene, standing still, I'll go from 65 fps to 15, then ten seconds later back to 65. I'm uncertain if this is a proton issue or not. I did find someone mentioning that it might help to shift PhysX to the GPU, but I'm not aware of a way to do that.

EDIT: Okay, I've been trying to pin this down with little success. Right now I'm going with the assumption that it's inconsistent thermal throttling due to the heat here, and not actually proton-related.

**Comment by ipr** [comment]
Looks like Proton experimental loads into game now, progress!

It does exit on a cutscene now. Previous attempt left system hanging, while now it just exits on entry to that cutscene. (There was just now update to experimental-version.)

**Comment by tgurr** [comment]
Works for me again as well now, even the cutscenes it seems, I experienced no crashes anymore so far. However there's something fishy going on with vsync, as soon as it's enabled the framerates are halfed (on nvm alt-tabbing seems to fix that so a game restart might probably as well).

Another issue I experienced is that Steam Cloud Save didn't restore my previous save game upon reinstalling Shadow Warrior 2 and creating a new prefix under compatdata, so I lost all my progress ☹️.

**Comment by kodatarule** [comment]
> Works for me again as well now, even the cutscenes it seems, I experienced no crashes anymore so far. However there's something fishy going on with vsync, as soon as it's enabled the framerates are halfed (on nvm alt-tabbing seems to fix that so a game restart might probably as well).
> 
> Another issue I experienced is that Steam Cloud Save didn't restore my previous save game upon reinstalling Shadow Warrior 2 and creating a new prefix under compatdata, so I lost all my progress frowning_face.

Game seems to be working just fine on latest proton experimental, but yes just like ninja gaiden master collections this one also has problems re-downloading a save from the steam cloud

**Comment by ipr** [comment]
Crash in cutscene with Proton experimental, divide by zero:
wine: Unhandled division by zero at address 0000000140319AAA (thread 0144), starting debugger...

[steam-324800.log](https://github.com/ValveSoftware/Proton/files/7556763/steam-324800.log)

**Comment by ipr** [comment]
With Proton 6.3-8 this now crashes before reaching main menu (NULL pointer, access violation). So that is worse than before.

[steam-324800.log](https://github.com/ValveSoftware/Proton/files/7600235/steam-324800.log)

**Comment by alasky17** [comment]
@ipr I think that Shadow Warrior 2 was hit by the same regression that broke eFootball PES 2021 etc and should be fixed with the next proton 6.3 release (it is already fixed in the recent experimental update today).  Edit: 6.3-8c release with the fix is live :)

**Comment by EpicureanGit** [comment]
OS: Slackware64-Current
GPU: NVIDIA GTX 1070, using driver version 495.46
Kernel: 5.15.17 
Using Proton Experimental: "Proton: 1642614452 experimental-6.3-20220119"
It crashes in the 2nd mission after using the 2nd key card and going through the doorway that it opens. 
PROTON_LOG=1
https://gist.github.com/EpicureanGit/1d647a334d828d25507749178b9ded7e

**Comment by EpicureanGit** [comment]
Tried: Proton 7.0-1 and Proton Experimental
It crashed at the same spot that I mentioned in my first post. There is a black screen and then crash.

**Comment by PredatorCZ** [comment]
Ran a new game, can confirm issue @EpicureanGit had, caused by `Unhandled exception: divide by zero in 64-bit code`
[steam-324800.log](https://github.com/ValveSoftware/Proton/files/8141854/steam-324800.log)

**Comment by ipr** [comment]
So far working with Proton 7.0 for me (crashed before). Mesa 21.3.5 and kernel 5.16.12.

**Comment by PredatorCZ** [comment]
I'm getting serious slowdowns. mangohud crashes, but I'm assuming something is killing worker threads, since slowdowns are tied to CPU events (physics,combat), when I stand still slowdowns are barely noticeable.
They appear to triggered during combat encounter or in menu (inventory management/crafting) after cca 30 minutes.
Tried low settings, disable nvapi. Using OpenGL will result in dark gamma and poor performance. It works fine in native Windows, so it's caused either by mesa drivers or dxvk.
I have tried current Experimental, GE 8, 6, 7, 8, 9 versions and the problem is across all of them.

[steam-324800.log](https://github.com/user-attachments/files/17850402/steam-324800.log)
